test/conformance/experimental_conformance_test.go

```diff
@@ -0,0 +1,134 @@
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/yaml"
+
+	"sigs.k8s.io/gateway-api/apis/v1alpha2"
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+	confv1a1 "sigs.k8s.io/gateway-api/conformance/apis/v1alpha1"
+	"sigs.k8s.io/gateway-api/conformance/tests"
+	"sigs.k8s.io/gateway-api/conformance/utils/flags"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+var (
+	cfg                 *rest.Config
+	k8sClientset        *kubernetes.Clientset
+	mgrClient           client.Client
+	implementation      *confv1a1.Implementation
+	conformanceProfiles sets.Set[suite.ConformanceProfileName]
+	skipTests           []string
+)
+
+func TestExperimentalConformance(t *testing.T) {
+	var err error
+	cfg, err = config.GetConfig()
+	if err != nil {
+		t.Fatalf("Error loading Kubernetes config: %v", err)
+	}
+	mgrClient, err = client.New(cfg, client.Options{})
+	if err != nil {
+		t.Fatalf("Error initializing Kubernetes client: %v", err)
+	}
+	k8sClientset, err = kubernetes.NewForConfig(cfg)
+	if err != nil {
+		t.Fatalf("Error initializing Kubernetes REST client: %v", err)
+	}
+
+	err = v1alpha2.AddToScheme(mgrClient.Scheme())
+	assert.NoError(t, err)
+	err = v1beta1.AddToScheme(mgrClient.Scheme())
+	assert.NoError(t, err)
+
+	// experimental conformance flags
+	conformanceProfiles = sets.New(
+		suite.HTTPConformanceProfileName,
+		suite.TLSConformanceProfileName,
+	)
+
+	// if some conformance profiles have been set, run the experimental conformance suite...
+	implementation, err = suite.ParseImplementation(
+		*flags.ImplementationOrganization,
+		*flags.ImplementationProject,
+		*flags.ImplementationURL,
+		*flags.ImplementationVersion,
+		*flags.ImplementationContact,
+	)
+	if err != nil {
+		t.Fatalf("Error parsing implementation's details: %v", err)
+	}
+
+	experimentalConformance(t)
+}
+
+func experimentalConformance(t *testing.T) {
+	t.Logf("Running experimental conformance tests with %s GatewayClass\n cleanup: %t\n debug: %t\n enable all features: %t \n conformance profiles: [%v]",
+		*flags.GatewayClassName, *flags.CleanupBaseResources, *flags.ShowDebug, *flags.EnableAllSupportedFeatures, conformanceProfiles)
+
+	cSuite, err := suite.NewExperimentalConformanceTestSuite(
+		suite.ExperimentalConformanceOptions{
+			Options: suite.Options{
+				Client:     mgrClient,
+				RestConfig: cfg,
+				// This clientset is needed in addition to the client only because
+				// controller-runtime client doesn't support non CRUD sub-resources yet (https://github.com/kubernetes-sigs/controller-runtime/issues/452).
+				Clientset:            k8sClientset,
+				GatewayClassName:     *flags.GatewayClassName,
+				Debug:                *flags.ShowDebug,
+				CleanupBaseResources: *flags.CleanupBaseResources,
+				SkipTests:            skipTests,
+				SupportedFeatures:    sets.Set[suite.SupportedFeature]{}.Insert(suite.HTTPRouteExtendedFeatures.UnsortedList()...),
```

Comment:
you probably want to add a lot more https://github.com/kubernetes-sigs/gateway-api/blob/8ab4e18ca73533793f8a5d4e8c4248626b4690bf/conformance/utils/suite/features.go#L195